pipeline_dp/pipeline_operations.py

```diff
@@ -131,6 +193,13 @@ def group_by_key(self, rdd, stage_name: str = None):
     def filter(self, rdd, fn, stage_name: str = None):
         return rdd.filter(fn)
 
+    def filter_by_key(self,
+                      rdd,
+                      public_partitions,
+                      data_extractors,
+                      stage_name: str = None):
+        pass
```

Comment:
Nit: raise NotImplementedError("filter_by_key is not implemented in SparkRDDOperations")

Reply:
Done.